src/usecases/uninstalled-use-case.ts

```diff
@@ -1,7 +1,28 @@
-import { connectInstallationRepository } from '../infrastructure/repositories';
+import { figmaService } from '../infrastructure/figma';
+import {
+	connectInstallationRepository,
+	figmaTeamRepository,
+} from '../infrastructure/repositories';
 
 export const uninstalledUseCase = {
 	execute: async (clientKey: string) => {
+		const connectInstallation =
+			await connectInstallationRepository.getByClientKey(clientKey);
+
+		const figmaTeams =
+			await figmaTeamRepository.findManyByConnectInstallationId(
+				connectInstallation.id,
+			);
+
+		await Promise.all(
+			figmaTeams.map((figmaTeam) =>
+				figmaService.deleteWebhook(
+					figmaTeam.webhookId,
+					figmaTeam.figmaAdminAtlassianUserId,
+				),
+			),
+		);
+
 		await connectInstallationRepository.deleteByClientKey(clientKey);
```

Comment:
What do you think about always deleting the `ConnectInstallation` (which will cascade delete `FigmaTeam` and `AssociatedDesign` records), even if deleting webhooks fails? Since uninstall events aren't retried, I don't think we'll get another opportunity to delete the `ConnectInstallation`, at least while don't have some queue implemented.

If we always delete the `ConnectInstallation`, we will also fail any received webhook events that we couldn't unsubscribe from. So hopefully Figma would delete the webhook when it never receives a successful response for the webhook event request.

Reply:
Excellent point, Justin! I was trying to design the use case for idempotency (in case the Figma team would like to add a queue) but I totally agree that we need to consider what we have right now (no queue and no retries).

I've updated the implementation and added a comment with more information about possible issues.